### ISSUE TITLE
all: be consistent about the term "embedded field"

### ISSUE THREAD

**Issue by mvdan** [golang/go#25684]
The spec very clearly calls fields with no names "embedded fields": https://golang.org/ref/spec#Struct_types

> A field declared with a type but no explicit field name is called an embedded field.

However, the Go standard library and documentation mixes that term with "anonymous field":

```
$ cd tip
$ git grep -i 'embedded field' | wc -l
83
$ git grep -i 'anonymous field' | wc -l
38
```

In particular, I found it in https://golang.org/pkg/go/ast/#Field:

    Names   []*Ident      // field/method/parameter names; or nil if anonymous field

Since the spec only mentions one, and it seems to dominate in usage anyway, I think we should try to be as consistent as possible. Specific cases have been fixed in the past, like https://github.com/golang/go/commit/b396d1143b3e717eb2828a101feeb8eb6810891b.

Of course, this doesn't include names that would break Go1 if changed - it is mainly meant for godocs, comments, and errors messages given by `go/*` and `cmd/*`. I presume that old changelogs, such as `doc/devel/weekly.html`, don't need to be changed.

/cc @griesemer @rsc @mdempsky

**Comment by griesemer** [comment]
@mvdan Yes, we probably want to be as consistent as possible. In the early days we used both terms; more recently we have started talking about 'embedded' fields because even though they are embedded they still have a name (the type name) and thus are not anonymous in the strict sense.

That said, note that ast.Fields are also used for parameters which indeed may be anonymous. So we want to be a little bit judicious before making unilateral changes throughout.

But feel free to send me CLs.

**Comment by mvdan** [comment]
> That said, note that ast.Fields are also used for parameters which indeed may be anonymous. So we want to be a little bit judicious before making unilateral changes throughout.

I'm not sure I understand; could you please give an example of a field that is embedded and anonymous, and an example of a field that is embedded but *not* anonymous?

**Comment by griesemer** [comment]
A Field is part of a FieldList. A FieldList is also used for function signatures (FuncType). Function parameters may be indeed anonymous and then Names may (will) be nil. In that case it makes sense to talk about anonymity (anonymous parameter). Fields are either embedded (originally called anonymous) or not.

In short, simply replacing "anonymous" with "embedded" everywhere will be confusing, too (there are no embedded parameters). When making those changes, the comments may need to be adjusted. That's all I'm saying.

**Comment by mvdan** [comment]
Ah, I understand now - thanks for the clarification. Would you prefer a few CLs grouping changes, or one single CL to clarify these?

**Comment by griesemer** [comment]
Depends on how many and where the fixes are. Use good judgement.

**Comment by mvdan** [comment]
Note that this issue was meant to track the inconsistencies across the entire repository, not just this occurrence in `go/ast`. As you can see in the numbers above, there are more than thirty other cases that should be checked, so I'd leave this issue open unless we are sure that all the others are correct.

**Comment by griesemer** [comment]
Reopening per @mvdan 's comment.
